### PR TITLE
docs(app-security): add docs for ionic-showcase (AEROGEAR-8694)

### PR DESCRIPTION
**Pull request by austincunningham**
## Motivation
https://issues.jboss.org/browse/AEROGEAR-8694

## What
Update the docs to for the ionic-showcase 

## Why
not documented 

## How
update docs

## Verification Steps
- clone the project 
- checkout my branch AEROGEAR-8694
- change directory into the root of the project 
- need node 8 
- install antora
```bash
npm i -g @antora/cli@1.0
npm i -g @antora/site-generator-default@1.0
mkdir build
```
- run the following script 
```bash
./bin/quick-build.sh local-site.yml
```
- checkout the following sections render as shown

showcase-docs

### CODE REVIEW

File: modules/ROOT/pages/_partials/showcase-apps/security.adoc [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+. View the results of the Device security checks.
+. Navigate to the {grafana-ui}, and view the metrics as described in xref:mobile-metrics.adoc[Analyzing App Usage]
+
+== {app-security-service}
+
+From the showcase app:
+
+. Press the *Security* menu item.
+. Press the *Application Trust* sub menu item.
+. Navigate to the Mobile Security Service link to set the enable/disable message as described in xref:app-security.adoc[using the Mobile Security Service]
```
Comment (craicoverflow) [inline modules/ROOT/pages/_partials/showcase-apps/security.adoc]: First two bullet points end in full stops but not the last two.

Comment (craicoverflow) [inline modules/ROOT/pages/_partials/showcase-apps/security.adoc]: Would it be possible to link directly to the relevant section of the page? This link sends the reader to the top of **Using the App Security Service**, and they will then have to find **Using the App Security Console**.

Also, the link says "using the Mobile Security Service", but this does not match the actual title of the page.

File: modules/ROOT/pages/_partials/showcase-apps/security.adoc [hunk 2]
```diff
@@ -0,0 +1,20 @@
+= Security Services
+
+== {device-security-service}
+
+
+From the showcase app:
+
+. Press the *Security* menu item.
```
Comment (jstaffor) [inline modules/ROOT/pages/_partials/showcase-apps/security.adoc]: I'd shorten the sentance (and apply this to the other sentances too) to :

" Click *Security* "

File: modules/ROOT/pages/_partials/showcase-apps/security.adoc [hunk 3]
```diff
@@ -0,0 +1,20 @@
+= Security Services
+
+== {device-security-service}
+
```
Comment (jstaffor) [inline modules/ROOT/pages/_partials/showcase-apps/security.adoc]: I'd add a line or two here describing what the user will achieve if the execute the steps below.

File: modules/ROOT/pages/_partials/showcase-apps/security.adoc [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+From the showcase app:
+
+. Press the *Security* menu item.
+. Press the *Device Trust* sub menu item.
+. View the results of the Device security checks.
+. Navigate to the {grafana-ui}, and view the metrics as described in xref:mobile-metrics.adoc[Analyzing App Usage]
+
+== {app-security-service}
+
```
Comment (jstaffor) [inline modules/ROOT/pages/_partials/showcase-apps/security.adoc]: I'd add a line or two here describing what the user will achieve if they execute the steps below.

File: modules/ROOT/pages/_partials/showcase-apps/security.adoc (hunk 1 above)
Comment (austincunningham) [reply modules/ROOT/pages/_partials/showcase-apps/security.adoc]: Unfortunately not , limitation of  asciidoc , links with underscore in them don't work, see https://github.com/asciidoctor/asciidoctor/issues/625 . So our `_partials` directory the issue here. Tried some of the workarounds with no joy.

On your second point links don't need to match the title of the page they are going to.

File: modules/ROOT/pages/_partials/showcase-apps/security.adoc (hunk 2 above)
Comment (austincunningham) [reply modules/ROOT/pages/_partials/showcase-apps/security.adoc]: I was matching the Auth section
image

No issue to changing , but would use press as it is a mobile app and click wouldn't apply.
```
. Press *Security*
. From the drop down press *Device Security*
```

File: modules/ROOT/pages/_partials/showcase-apps/security.adoc (hunk 1 above)
Comment (finp) [reply modules/ROOT/pages/_partials/showcase-apps/security.adoc]: _partials is only used for includes, should never affect linking.
I'm not sure what the issue is @austincunningham , but reach out if you wanna discuss

Comment (finp) [reply modules/ROOT/pages/_partials/showcase-apps/security.adoc]: FYI: nothing in _partials is published, ie unless it's included in a file that exists in ../pages, no html is produced.

Comment (austincunningham) [reply modules/ROOT/pages/_partials/showcase-apps/security.adoc]: That helps thanks @finp

File: modules/ROOT/pages/_partials/showcase-apps/security.adoc (hunk 2 above)
Comment (austincunningham) [reply modules/ROOT/pages/_partials/showcase-apps/security.adoc]: usage is 
https://user-images.githubusercontent.com/16667688/56582990-7611ed00-65d0-11e9-8e60-23d7692c9c56.gif

Comment (finp) [reply modules/ROOT/pages/_partials/showcase-apps/security.adoc]: I think 'Press' is good for highlighting the mobile UI, but haven't researched if there are style issues with that.

**Pull request merged**